packages/javascript-sdk/vite.config.ts

```diff
@@ -1,19 +1,46 @@
 /// <reference types='vitest' />
 import { defineConfig } from 'vite';
-
-import { nxViteTsPaths } from '@nx/vite/plugins/nx-tsconfig-paths.plugin';
+import dts from 'vite-plugin-dts';
+// import { nxViteTsPaths } from '@nx/vite/plugins/nx-tsconfig-paths.plugin';
+import { copyFileSync } from 'fs';
 
 export default defineConfig({
   root: __dirname,
   cacheDir: '../../node_modules/.vite/packages/javascript-sdk',
 
-  plugins: [nxViteTsPaths()],
-
   // Uncomment this if you are using workers.
   // worker: {
   //  plugins: [ nxViteTsPaths() ],
   // },
-
+  build: {
+    lib: {
+      entry: 'src/index.ts',
+      name: 'javascript-sdk',
+      formats: ['es', 'cjs'],
+      fileName(format, name) {
+        return `${name}.${format === 'cjs' ? 'js' : 'mjs'}`;
+      },
+    },
+    rollupOptions: {
+      output: {
+        dir: 'dist/packages/javascript-sdk',
+        preserveModules: true,
+        preserveModulesRoot: 'packages/javascript-sdk',
+      },
+    },
+  },
+  plugins: [
+    dts({
+      copyDtsFiles: true,
+      outDir: '../../dist',
+      rollupTypes: false,
+      insertTypesEntry: true,
+      tsconfigPath: './tsconfig.lib.json',
+      afterBuild: (files) => {
+        return files.forEach((value, key) => copyFileSync(key, key.replace('.ts', '.mts')));
```

Comment:
I see we are moving more towards the `.mts`/`.mjs`. Can you elaborate as to why?

Reply:
This is so we have both mts and d.ts files.

when you publish dual module system you need to have both sets of types. if we don't have mts, then the mjs files won't have types properly. Because we have commonjs, d.ts respects the commonjs, but the mjs files need mts types.